src/main.rs

```diff
@@ -56,43 +26,35 @@ fn main() -> std::io::Result<()> {
     // destruction of it
     let runtime = create_search_runtime(settings.storage.performance.max_search_threads)
         .expect("Can't create runtime.");
-    let handle = runtime.handle().clone();
-
-    actix_web::rt::System::new().block_on(async {
-        let toc = TableOfContent::new(&settings.storage, handle);
-
-        for collection in toc.all_collections() {
-            info!("loaded collection: {}", collection);
-        }
-
-        let toc_data = web::Data::new(toc);
-
-        HttpServer::new(move || {
-            App::new()
-                .wrap(Logger::default())
-                .app_data(toc_data.clone())
-                .app_data(Data::new(
-                    web::JsonConfig::default()
-                        .limit(32 * 1024 * 1024)
-                        .error_handler(json_error_handler),
-                )) // 32 Mb
-                .service(index)
-                .service(get_collections)
-                .service(update_collections)
-                .service(get_collection)
-                .service(update_points)
-                .service(get_point)
-                .service(get_points)
-                .service(scroll_points)
-                .service(search_points)
-                .service(recommend_points)
-        })
-        // .workers(4)
-        .bind(format!(
-            "{}:{}",
-            settings.service.host, settings.service.port
-        ))?
-        .run()
-        .await
-    })
+    let search_runtime_handle = runtime.handle().clone();
+
+    let toc = TableOfContent::new(&settings.storage, search_runtime_handle);
+    for collection in toc.all_collections() {
+        info!("loaded collection: {}", collection);
+    }
+    let toc_arc = Arc::new(toc);
+
+    let mut handles: Vec<JoinHandle<Result<(), Error>>> = vec![];
+
+    #[cfg(feature = "web")]
+    {
+        let toc_arc = toc_arc.clone();
+        let settings = settings.clone();
+        let handle = thread::spawn(move || actix::init(toc_arc, settings));
+        handles.push(handle);
+    }
+    #[cfg(feature = "grpc")]
+    {
+        let toc_arc = toc_arc.clone();
+        let settings = settings.clone();
+        let handle = thread::spawn(move || tonic::init(toc_arc, settings));
+        handles.push(handle);
+    }
+
+    for handle in handles.into_iter() {
+        handle.join().expect("Couldn't join on the thread")?;
+    }
+    drop(toc_arc);
+    drop(settings);
```

Comment:
Won't it drop automatically?

Reply:
It would, of course. These 2 line serve to purposes:
- Make the lifecycle of toc and settings objects explicit
- Avoid clippy warnings of type "unused variable" and "non-requred" clone with any combination of the feature flags. For examle, clippy suggests to avoid:
```
        let toc_arc = toc_arc.clone();
        let settings = settings.clone();
```
line in the last enabled feature block. It considers that ``toc_arc`` and ``settings`` are not used in the rest of the ``main`` function, so it suggests passing ownership of them to tonic. However, Clippy does not consider the lifetime of the object and passing it ownership would end up with an error of deletion of the runtime in the async context. Having drop expilidly states that the objects are deleted in the main function (becuase ``join()`` ensures the runtime is deleted). It also ensures that, when new feature is added in future, the programmer would not be confused by the same clippy warning